sdk/core/core-http/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -84,8 +99,24 @@ export class BearerTokenAuthenticationPolicy extends BaseRequestPolicy {
   private async getToken(options: GetTokenOptions): Promise<string | undefined> {
     let accessToken = this.tokenCache.getCachedToken();
     if (accessToken === undefined) {
-      accessToken = (await this.credential.getToken(this.scopes, options)) || undefined;
-      this.tokenCache.setCachedToken(accessToken);
+      // Waiting for the next refresh (or forcefully triggering a new refresh)
+      // only if the cache is unable to retrieve the access token,
+      // which means that it has expired, or it has never been set.
+      const refreshPromise = this.tokenRefresher.forcedRefresh(options);
+      if (refreshPromise !== null) {
+        accessToken = await refreshPromise;
+      }
+    } else {
+      // If we still have a cached access token,
+      // then attempt to refresh without waiting.
+      const refreshPromise = this.tokenRefresher.refresh(options);
+      // If the tokenRefresher returned null, some other refresh is happening already.
+      // if this is a new refresh, we set it up to update the cachedToken once it finishes.
+      if (refreshPromise !== null) {
+        refreshPromise.then((accessToken: AccessToken | undefined) => {
```

Comment:
why does this fork the promise chain instead of using await?

Reply:
the idea is to not wait for this to happen, that's the main point of this feature: this should happen in the background.